sdk/core/core/src/az_span.c

```diff
@@ -144,89 +144,57 @@ AZ_NODISCARD az_result az_span_find(az_span source, az_span target, az_span* out
  *    - a byte in `target` is different than `source`, in the expected corresponding position;
  *    - the loop has reached the end of `source` (and there are still remaing bytes of `target` to be checked).
  */
-  AZ_PRECONDITION_VALID_SPAN(source, 1, false);
-  AZ_PRECONDITION_VALID_SPAN(target, 1, false);
-  AZ_PRECONDITION_NOT_NULL(out_span);
 
-  uint8_t* source_ptr = az_span_ptr(source);
   int32_t source_length = az_span_length(source);
-  uint8_t* target_ptr = az_span_ptr(target);
   int32_t target_length = az_span_length(target);
+  const int32_t target_not_found = -1;
 
-  // This loop traverses `source` position by position (step 1.)
-  for (int i = 0; i < source_length; i++)
-  {
-    // This is the check done in step 1. above.
-    if (source_ptr[i] == target_ptr[0])
-    {
-      // The condition in step 2. has been satisfied.
-      int j;
-      // This is the loop defined in step 3.
-      // The loop must be broken if it reaches the ends of `target` (step 3.) OR `source` (step 5.).
-      for (j = 1; j < target_length && (i + j) < source_length; j++)
-      {
-        // Condition defined in step 5.
-        if (source_ptr[i + j] != target_ptr[j])
-        {
-            break;
-        }
-      }
- 
-      if (j == target_length)
-      {
-        // All bytes in `target` have been checked and matched the corresponding bytes in `source` (from the start point `i`),
-        // so this is indeed an instance of `target` in that position of `source` (step 4.).
-        // Here we create a span in `source` to map the occurrence of `target` in `source` (pardon the redundancy of words, done for explicit clarity).
-        *out_span = az_span_init(source_ptr + i, target_length, target_length);
-
-        return AZ_OK;
-      }
-    }
+  if (target_length == 0)
+  { 
+    return 0;
   }
-
-  // If the function hasn't returned before, all positions 
-  // of `source` have been evaluated but `target` could not be found.
-  return AZ_ERROR_ITEM_NOT_FOUND;
-}
-
-AZ_NODISCARD az_span az_span_token(az_span source, az_span delimiter, az_span* out_remainder)
-{
-  AZ_PRECONDITION_VALID_SPAN(delimiter, 1, false);
-  AZ_PRECONDITION_NOT_NULL(out_remainder);
-
-  if (az_span_is_content_equal(source, AZ_SPAN_NULL))
+  else if (source_length == 0 || source_length < target_length) 
```

Comment:
Sorry, I just realized these conditions can be merged given target_length is greater than 0 at this point (because of the check above). And if `source_length == 0` when `target_length > 0`, `source_length < target_length` is already true.

Please validate, but I believe this should work:
```suggestion
  else if (source_length < target_length) 
```

Reply:
Absolutely correct. Fixed.